<!--
 Software License Agreement (BSD License)
 Copyright Â© 2022-2023 belongs to Shadow Robot Company Ltd.
 All rights reserved.
 Redistribution and use in source and binary forms, with or without modification,
 are permitted provided that the following conditions are met:
   1. Redistributions of source code must retain the above copyright notice,
      this list of conditions and the following disclaimer.
   2. Redistributions in binary form must reproduce the above copyright notice,
      this list of conditions and the following disclaimer in the documentation
      and/or other materials provided with the distribution.
   3. Neither the name of Shadow Robot Company Ltd nor the names of its contributors
      may be used to endorse or promote products derived from this software without
      specific prior written permission.
 This software is provided by Shadow Robot Company Ltd "as is" and any express
 or implied warranties, including, but not limited to, the implied warranties of
 merchantability and fitness for a particular purpose are disclaimed. In no event
 shall the copyright holder be liable for any direct, indirect, incidental, special,
 exemplary, or consequential damages (including, but not limited to, procurement of
 substitute goods or services; loss of use, data, or profits; or business interruption)
 however caused and on any theory of liability, whether in contract, strict liability,
 or tort (including negligence or otherwise) arising in any way out of the use of this
 software, even if advised of the possibility of such damage.
-->

<launch>

  <!-- where to print the node output -->
  <arg name="output" default="screen"/>

  <!-- if the robot should be controlled in simulation -->
  <arg name="sim" default="true"/>
  
  <!-- model gravity -->
  <arg name="model_gravity" default="true"/>
  
  <!-- set model gravity -->
  <arg name="model_name_suffix" default="static" if="$(eval not model_gravity)"/>
  <arg name="model_name_suffix" value="dynamic"  if="$(eval     model_gravity)"/>
  
  <!-- prop attributes -->
  <arg name="prop" default="cube" />
  <arg name="model_path" default="$(find shadow_hand)/../models/" />
  <arg name="prop_pose_1" default="-x 0.5 -y 1.0 -z 0.55 -R 0.0 -P 0.0 -Y 0.0" />
  <arg name="prop_pose_2" default="-x -0.5 -y 1.0 -z 0.55 -R 0.0 -P 0.0 -Y 0.0" />
  
  <!-- spawn prop -->
  <group unless="$(eval arg('prop') == 'False')">
    <node name="spawn_prop_1" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg model_path)$(arg prop)/model_$(arg model_name_suffix).sdf -sdf $(arg prop_pose_1) -model $(arg prop)_1"/>
    <node name="spawn_prop_2" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg model_path)$(arg prop)/model_$(arg model_name_suffix).sdf -sdf $(arg prop_pose_2) -model $(arg prop)_2"/>
    <node name="spawn_prop_3" pkg="gazebo_ros" type="spawn_model" respawn="false" output="screen" args="-file $(arg model_path)target/model_static.sdf -sdf -x 0.5 -y 0.5 -z 0.5 -model target"/>
  
  </group>

  <!-- <arg name="physics_engine" default="dart" /> -->

  <!-- <node name="gazebo_34" pkg="gazebo_ros" type="gazebo" respawn="false" output="screen">
    <param name="physics_engine" value="$(arg physics_engine)" />
  </node> -->

  <!-- robot_control_demo node -->
  <!-- <node pkg="shadow_hand" name="debug" type="debug.py" output="$(arg output)" /> -->
  <node pkg="shadow_hand" name="robot_control_demo" type="robot_control_demo.py" output="$(arg output)" />

  <arg name="initial_z" default="0.5"/>

  <!-- This launch file is for the right ur10 arm and a shadow hand -->
  <include file="$(find sr_robot_launch)/launch/sr_right_ur10arm_hand.launch">
    <arg name="sim" value="$(arg sim)" />
    <arg name="initial_z" value="$(arg initial_z)" />
  </include>

  <!-- <node name="reset_gazebo" pkg="rosservice" type="rosservice" args="call /gazebo/reset_world " output="$(arg output)"/> -->

</launch>
